index.html

```diff
@@ -75,12 +75,13 @@ <h2>
 
           const unreadCount = await getUnreadMailCount();
 
-          // Try to set the app badge.
+          // Set the app badge - this will always succeed if the API is supported.
+          // The operating system controls whether the badge is actually displayed.
           try {
             await navigator.setAppBadge(unreadCount);
           } catch (e) {
-            // The badge is not supported, or the user has prevented the app
-            // from setting a badge.
+            // Errors are rare and typically indicate API unavailability
+            // or security restrictions, not permission issues.
```

Comment:
nit: think its probably worth removing typically here
```suggestion
            // Errors are rare and indicate API unavailability security restrictions, not permission
            // issues.
```